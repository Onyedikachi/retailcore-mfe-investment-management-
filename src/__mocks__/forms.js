export const countries = [
  {
    id: 1,
    name: "Nigeria",
    value: "Nigeria",
  },
  {
    id: 2,
    name: "Ghana",
    value: "Ghana",
  },
];

export const states = [
  {
    id: 1,
    name: "Lagos",
    value: "Lagos",
  },
  {
    id: 2,
    name: "Ibadan",
    value: "Ibadan",
  },
];

export const LGAs = [
  {
    id: 1,
    name: "Abuja",
    value: "Abuja",
  },
  {
    id: 2,
    name: "Asokoro",
    value: "Asokoro",
  },
];
