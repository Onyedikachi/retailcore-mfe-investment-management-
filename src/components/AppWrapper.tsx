import { Provider } from "react-redux";
import { PersistGate } from "redux-persist/integration/react";
import { persistStore } from "redux-persist";
import { ToastContainer, Zoom } from "react-toastify";
import { store } from "../config/store";

function AppWrapper({ children }) {
  let persistor = persistStore(store());

  return (
    <div data-testid="app">
      <Provider data-testid="store" store={store()}>
        <PersistGate loading={<div>loading...</div>} persistor={persistor}>
          <ToastContainer
            position="top-right"
            autoClose={4000}
            draggable={false}
            // icon={false}
            transition={Zoom}
          />
          {children}
        </PersistGate>
      </Provider>
    </div>
  );
}

export default AppWrapper;
