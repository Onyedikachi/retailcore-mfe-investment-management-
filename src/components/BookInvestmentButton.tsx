import React, { useState } from "react";
import OutsideClickHandler from "react-outside-click-handler";
import { BookInvestmentOptions, IBookInvestmentOptions } from "@app/constants";
import { useNavigate } from "react-router-dom";

type BookInvestmentButton = {
  children?: any;
  disabled?: boolean;
};

export default function BookInvestmentButton({
  children,
  disabled,
}: BookInvestmentButton) {
  const navigate = useNavigate();
  const [isOpen, setIsOpen] = useState(false);

  const goToUrl = (url) => {
    if (disabled) return;
    navigate(url);

    setIsOpen(false);
  };

  return (
    <div>
      <OutsideClickHandler
        onOutsideClick={() => {
          setIsOpen(false);
        }}
      >
        <div className="relative">
          <div onKeyUp={() => {}} onClick={() => setIsOpen(true)}>{children}</div>
          {isOpen && (
            <div className="top-0 absolute w-[286px] z-10 bg-white rounded-b-lg shadow-[0px_0px_4px_0px_rgba(0,0,0,0.25)] pt-[1px]">
              <ul>
                {BookInvestmentOptions?.map((item: IBookInvestmentOptions) => (
                  <li className="group" key={item.title}>
                    <button
                      disabled={disabled}
                      onClick={() => {
                        goToUrl(item.url);
                      }}
                      className={`disabled:opacity-50 disabled:cursor-not-allowed w-full z-10 relative text-base text-[#636363] capitalize pl-6 pr-4 flex justify-between items-center py-3 group-last:rounded-b-lg hover:bg-[#F9E5E5] cursor-pointer`}
                    >
                      {item.title}
                    </button>
                  </li>
                ))}
              </ul>
            </div>
          )}
        </div>
      </OutsideClickHandler>
    </div>
  );
}
