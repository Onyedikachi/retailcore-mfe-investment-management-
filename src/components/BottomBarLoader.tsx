import React from "react";

const BottomBarLoader = () => {
  return (
    <div
      data-testid="bottom-bar-loader"
      className="w-full  p-2 flex justify-center items-center"
    >
      <div className="spinner-border h-10 w-10 border-t border-danger-500 rounded-full animate-spin"></div>
    </div>
  );
};

export default BottomBarLoader;
