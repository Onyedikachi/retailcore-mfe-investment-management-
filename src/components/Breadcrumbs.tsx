import React from "react";
import { BsArrowLeft } from "react-icons/bs";
import { useNavigate } from "react-router-dom";

interface BreadcrumbItem {
  id: string | number;
  title: string;
  url: string;
}

interface BreadcrumbsProps {
  links: BreadcrumbItem[];
}

export default function Breadcrumbs({
  links,
}: BreadcrumbsProps): React.ReactElement {
  const navigate = useNavigate();
  return (
    <div className="flex items-center gap-x-2" data-testid="breadcrumbs">
      <button
        className="py-1 px-1 h-8 w-8 rounded flex items-center justify-center outline-none bg-transparent hover:bg-[#F9F2F2]"
        onClick={() => navigate(-1)}
        data-testid="back button"
      >
        <BsArrowLeft className="text-[#636363]" />{" "}
      </button>
      <ul className="flex gap-x-8 uppercase items-center">
        {links.map((item: BreadcrumbItem, idx: number) => (
          <li
            key={`${idx.toString()}-index`}
            onMouseOver={() => { }}
            onClick={() => navigate(item.url)}
            data-testid={item.title}
            className={`first-of-type:hover:bg-[#F9F2F2] ${item.url !== "#" ? "hover:bg-[#F9F2F2] cursor-pointer" : ""
              } px-2 py-1 rounded first-of-type:cursor-pointer relative text-[#8F8F8F] text-base font-medium last-of-type:text-[#636363] after:absolute after:content-[''] after:right-[-16px] after:top-1/2 after:translate-y-[-50%] after:w-2 after:h-2 after:border-b-2 after:border-r-2 after:border-[#636363] after:-rotate-45 last-of-type:after:content-none`}
          >
            <span>{item.title}</span>
          </li>
        ))}
      </ul>
    </div>
  );
}
