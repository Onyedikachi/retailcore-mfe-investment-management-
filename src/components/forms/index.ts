export { default as Checkbox } from "./Checkbox";
export { default as ComboSelect } from "./ComboSelect";
export { default as DateSelect } from "./DateSelect";
export { default as FormUpload } from "./FormUpload";
export { default as MultiSelect } from "./MultiSelect";
export { default as Select } from "./Select";
export { default as CustomComboSelect } from "./CustomComboSelect";
export { default as BorderlessSelect } from "./BorderlessSelect";
export { default as CustomInput } from "./CustomInput";
export { default as GlInput } from "./GlInput";
export { default as FormDate } from "./FormDate";
export { default as MinMaxInput } from "./MinMaxInput";
export { default as InfoLabel } from "./InfoLabel";
export {default as RedDot} from './RedDot'