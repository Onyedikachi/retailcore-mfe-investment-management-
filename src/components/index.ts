export { default as Breadcrumbs } from "./Breadcrumbs";
export { default as DropDown } from "./DropDown";
export { default as Button } from "./Button";
export { default as Loader } from "./Loader";
export { default as MultiRangeSlider } from "./MultiRangeSlider";
export { default as ProcessingStatusSlider } from "./ProcessingStatusSlider";
export { default as TopBar } from "./TopBar";
export { default as StatusCard } from "./StatusCard";
export { default as TableComponent } from "./investment/dashboard/TableComponent";
export { default as QuickLinks } from "./QuickLinks";
export { default as FormStepComponent } from "./FormStepComponent";
export { default as ProductSearch } from "./ProductSearch";
