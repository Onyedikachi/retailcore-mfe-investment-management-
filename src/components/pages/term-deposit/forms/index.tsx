export { default as ProductInformation } from "./product-information";
export { default as CustomerEligibilityCriteria } from "./customer-eligibilty-criteria";
export { default as AccountingEntriesAndEvents } from "./accounting-entries-and-events";
export { default as LiquiditySetup } from "./liquidity-setup";
export { default as PricingConfig } from "./pricing-config";
export { default as ToggleInputChildren } from "./toggle-input-children";
export { default as EntriesAndEventsSearch } from "./entries-and-events-search";
