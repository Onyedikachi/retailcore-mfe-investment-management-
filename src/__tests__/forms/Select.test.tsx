import React from "react";
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import { Select } from "../../components/forms";
import userEvent from "@testing-library/user-event";

const options = [
  { id: 1, text: "Option 1", value: "option 1" },
  { id: 2, text: "Option 2", value: "option 2" },
];

function handleSelected(value: string) {}
describe("Select", () => {
  it("renders without crashing", () => {
    render(<Select options={options} handleSelected={handleSelected} />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <Select options={options} handleSelected={handleSelected} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("shows the selected option", () => {
    const handleSelected = jest.fn();
    const { getByText } = render(
      <Select options={options} handleSelected={handleSelected} />
    );
    expect(getByText("Option 1")).toBeInTheDocument();
  });

  it("opens the dropdown when clicked", async () => {
    render(<Select options={options} handleSelected={handleSelected} />);

    const toggle = screen.getByRole("button");

    userEvent.click(toggle);

    await screen.findByText("Option 1");

    expect(screen.queryByText("Option 1")).toBeInTheDocument();
  });

  it("calls onChange when option selected", async () => {
    const handleChange = jest.fn();

    render(
      <Select
        options={options}
        onChange={handleChange}
        handleSelected={handleSelected}
      />
    );
    const toggle = screen.getByRole("button");

    fireEvent.click(toggle);

    const option = await screen.findByText("Option 2");
    const optionElement1 = screen.getByRole("option", { name: "Option 1" });
    const optionElement2 = screen.getByRole("option", { name: "Option 2" });
    expect(optionElement1).toHaveClass(
      "relative cursor-pointer select-none py-2 px-6 text-[#636363] hover:bg-gray-50 bg-red-50"
    );
    expect(optionElement2).toHaveClass(
      "relative cursor-pointer select-none py-2 px-6 text-[#636363] hover:bg-gray-50"
    );
    const ele = (await screen.findByText("Option 2")) as HTMLElement;
    expect(ele).toHaveClass("font-normal");

    fireEvent.click(option);

    //  expect(handleChange).toHaveBeenCalledTimes(1);
  });
});
