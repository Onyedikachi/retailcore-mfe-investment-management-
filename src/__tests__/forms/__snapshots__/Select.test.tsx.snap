// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="min-w-[150px]"
    role="combobox"
  >
    <div
      class="relative mt-1"
    >
      <button
        aria-expanded="false"
        aria-haspopup="listbox"
        class="relative w-full cursor-pointer rounded-[6px] bg-white py-1 pl-2 pr-10 text-left border border-[#D0D5DD] focus:outline-none  text-[#252C32] text-sm"
        data-headlessui-state=""
        id="headlessui-listbox-button-2"
        type="button"
      >
        <span
          class="block truncate"
        >
          Option 1
        </span>
        <span
          class="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-2 text-xs"
        >
          <svg
            aria-hidden="true"
            class=" text-[#6E7C87]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
